src/main/java/org/openrewrite/java/spring/boot2/UnnecessarySpringExtension.java

```diff
@@ -92,13 +100,41 @@ public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, Ex
                                     c = (J.ClassDeclaration) new RemoveAnnotation(EXTEND_WITH_SPRING_EXTENSION_ANNOTATION_PATTERN)
                                             .getVisitor().visit(c, ctx, getCursor().getParentOrThrow());
                                     assert c != null;
-                                    maybeRemoveImport("org.springframework.test.context.junit.jupiter.SpringExtension");
-                                    maybeRemoveImport("org.junit.jupiter.api.extension.ExtendWith");
+                                    maybeRemoveImport(SPRING_EXTENSION);
+                                    maybeRemoveImport(EXTEND_WITH);
                                     return super.visitClassDeclaration(c.withBody(classDecl.getBody()), ctx);
                                 }
                             }
+                            return super.visitClassDeclaration(classDecl, ctx);
+                        }
+                        return classDecl;
+                        //return super.visitClassDeclaration(classDecl, ctx);
+                    }
+
+                    @Override
+                    public J.Annotation visitAnnotation(J.Annotation annotation, ExecutionContext ctx) {
+                        if (EXTEND_WITH_MATCHER.matches(annotation)) {
+
+                            Expression expression = annotation.getArguments().get(0);
+
+                            if (expression instanceof J.NewArray) {
+                                List<Expression> collected = ((J.NewArray) expression).getInitializer().stream()
+                                        .filter(e -> !isSpringExtension(e))
+                                        .collect(Collectors.toList());
+                                Expression newArgument = ((J.NewArray) expression).withInitializer(collected);
+                                J.Annotation newAnnotation = annotation.withArguments(Collections.singletonList(newArgument));
```

Comment:
can we change this to use `ListUtils.map()`, with a mapper that uses `return null` where an element should be dropped?

Reply:
https://github.com/openrewrite/rewrite-spring/pull/705/commits/3c975cb63f2bc7c39ff8012a6acf4bd83be7903f